blockchain/chain/src/store/chain_store.rs

```diff
@@ -301,6 +301,54 @@ where
         }
     }
 
+    pub async fn validate_tipset_checkpoints(
+        &self,
+        from: Arc<Tipset>,
+        to: ChainEpoch,
+        network: String,
+    ) -> Result<(), Error> {
+        info!(
+            "Validating {} tipset checkpoint hashes from: {}",
+            network,
+            from.epoch()
+        );
+
+        if to > from.epoch() {
+            return Err(Error::Other(format!(
+                "Received tipset with height ({}) greater than start ({}) point",
+                to,
+                from.epoch()
+            )));
+        }
+
+        let mut hashes = checkpoint_tipsets::get_tipset_hashes(&network);
+
+        let mut ts = from;
+        let tipset_hash = checkpoint_tipsets::tipset_hash(ts.key());
+        hashes.remove(&tipset_hash);
+
+        loop {
+            let pts = self.chain_index.load_tipset(ts.parents()).await?;
+            let tipset_hash = checkpoint_tipsets::tipset_hash(ts.key());
+            hashes.remove(&tipset_hash);
+
+            if to == pts.epoch() {
+                break;
+            }
+
+            ts = pts;
+        }
+
+        if !hashes.is_empty() {
+            Err(Error::Other(format!(
+                "Found tipset hash(es) on {network} that are no longer valid: {:?}",
+                hashes
+            )))
+        } else {
+            Ok(())
+        }
```

Comment:
At this point we've only validated that the current blockchain contains all the known checkpoints. We also want to make sure that the genesis CID of the chain actually matches the genesis CID of the checkpoints.

Reply:
Added genesis check.